test-tests/shared/src/test/scala/zio/test/TestAspectSpec.scala

```diff
@@ -0,0 +1,120 @@
+            after  <- ref.get
+          } yield {
+            assert(result, isTrue) &&
+            assert(after, equalTo(-1))
+          }
+        },
+        testM("js applies test aspect only on ScalaJS") {
+          for {
+            ref    <- Ref.make(false)
+            spec   = test("test")(assert(true, isTrue)) @@ js(after(ref.set(true)))
+            _      <- execute(spec)
+            result <- ref.get
+          } yield if (TestPlatform.isJS) assert(result, isTrue) else assert(result, isFalse)
+        },
+        testM("jsOnly runs tests only on ScalaJS") {
+          val spec   = test("Javascript-only")(assert(TestPlatform.isJS, isTrue)) @@ jsOnly
+          val result = if (TestPlatform.isJS) succeeded(spec) else ignored(spec)
+          assertM(result, isTrue)
+        },
+        testM("jvm applies test aspect only on jvm") {
+          for {
+            ref    <- Ref.make(false)
+            spec   = test("test")(assert(true, isTrue)) @@ jvm(after(ref.set(true)))
+            _      <- execute(spec)
+            result <- ref.get
+          } yield assert(if (TestPlatform.isJVM) result else !result, isTrue)
+        },
+        testM("jvmOnly runs tests only on the JVM") {
+          val spec   = test("JVM-only")(assert(TestPlatform.isJVM, isTrue)) @@ jvmOnly
+          val result = if (TestPlatform.isJVM) succeeded(spec) else ignored(spec)
+          assertM(result, isTrue)
+        },
+        test("failure makes a test pass if the result was a failure") {
+          assert(throw new java.lang.Exception("boom"), isFalse)
+        } @@ failure,
+        test("failure makes a test pass if it died with a specified failure") {
+          assert(throw new NullPointerException(), isFalse)
+        } @@ failure(failsWithException[NullPointerException]),
+        test("failure does not make a test pass if it failed with an unexpected exception") {
+          assert(throw new NullPointerException(), isFalse)
+        } @@ failure(failsWithException[IllegalArgumentException])
+          @@ failure,
+        test("failure does not make a test pass if the specified failure does not match") {
+          assert(throw new RuntimeException(), isFalse)
+        } @@ failure(
+          isCase[TestFailure[String], Cause[String]]("Runtime", {
+            case TestFailure.Runtime(e) => Some(e); case _ => None
+          }, equalTo(Cause.fail("boom")))
+        ) @@ failure,
+        test("failure makes tests pass on any assertion failure") {
+          assert(true, equalTo(false))
+        } @@ failure,
+        test("failure makes tests pass on an expected assertion failure") {
+          assert(true, equalTo(false))
+        } @@ failure(
+          isCase[TestFailure[Any], Any](
+            "Assertion",
+            { case TestFailure.Assertion(result) => Some(result); case _ => None },
+            anything
+          )
+        ),
+        /*doesn't compile, aspects do seem to work as intended below though...
```

Comment:
Can this be deleted? Looks like it is compiling in the test below unless I am missing something. Is there a type inference issue with test aspects we still need to address?

Reply:
Never mind - user error! I was using `test` instead of `testM`

Intellij still gets confused with the types but the compiler has it sorted